superset/assets/src/components/TableSelector.jsx

```diff
@@ -111,7 +111,11 @@ export default class TableSelector extends React.PureComponent {
     if (data.result.length === 0) {
       this.props.handleError(t("It seems you don't have access to any database"));
     }
-    return data.result;
+    return data.result.map(row => ({
+      ...row,
+      // label is used for the typeahead
+      label: row.backend + ' ' + row.database_name,
```

Comment:
[nit] would suggest doing this via string interpolation with `${row.backend} ${row.database_name}`

Reply:
Ah, nice! Will do that.